GERTi/GERTiClient.lua

```diff
@@ -201,57 +292,71 @@ handler["RemoveNeighbor"] = function (eventName, receivingModem, sendingModem, p
 end
 
 handler["RegisterNode"] = function (eventName, receivingModem, sendingModem, port, distance, code, origination, tier, serialTable)
+	-- THIS IS COMPLETELY IGNORED. Until I have more knowledge about the organization, I won't touch this,
+	--  but FIX THIS LATER! Also see RegisterComplete and the stuff that happened to ResolveAddress - 20kdc
 	transmitInformation(neighbors[1]["address"], neighbors[1]["port"], "RetrieveAddress", origination, tier, serialTable)
 	local _, _, _, port, _, payload = event.pull(5, "modem_message")
 	if payload ~= nil then
 		return transmitInformation(sendingModem, port, payload)
 	end
 end
 
+handler["RegisterComplete"] = function (eventName, receivingModem, sendingModem, port, distance, code, targetMA, iResponse)
+	if receivingModem == targetMA then
+		-- This is us!
+		iAddress = iResponse
+	end
+	-- The code that goes in the 'else' depends on what happens to RegisterNode
+end
+
 handler["RETURNSTART"] = function (eventName, receivingModem, sendingModem, port, distance, code, tier)
 	-- Store neighbor based on the returning tier
-	return storeNeighbors(eventName, receivingModem, sendingModem, port, distance, tier)
+	storeNeighbors(eventName, receivingModem, sendingModem, port, distance, tier)
+end
+
+handler["ResolveAddress"] = function (eventName, receivingModem, sendingModem, port, distance, code, gAddress)
+	addWaitingAddress(gAddress, function(response)
+		-- this indentation is weird.
+		-- Anyway, this is the callback to forward the resolution
+		transmitInformation(sendingModem, port, "AddressResolution", gAddress, response)
```

Comment:
Unsure of why we are switching terms here. Resolve Address should be used up the chain, and temporary handlers added to allow everything to be quickly forwarded back down.